lib/installation-proxy/index.js

```diff
@@ -0,0 +1,56 @@
+const INSTALLATION_PROXY_SERVICE_NAME = 'com.apple.mobile.installation_proxy';
+
+class InstallationProxyService {
+  constructor (plistService) {
+    this.plistService = plistService;
+  }
+
+  async listApplications (applicationType = 'User') {
+    const request = {
+      Command: 'Browse',
+      ClientOptions: {
+        ApplicationType: applicationType
+      }
+    };
+
+    this.plistService.sendPlist(request);
+    const messages = await this.waitMessageCompletion();
+    let map = {};
+    for (const message of messages) {
+      for (const app of message.CurrentList) {
+        map[app.CFBundleIdentifier] = app;
+      }
+    }
+    return map;
+  }
+
+  async waitMessageCompletion () {
+    let messages = [];
+    for (;;) {
```

Comment:
I'd rather loop to Number.MAX_SAFE_INTEGER, just to make sure we are never going to loop forever